### PR TITLE
Build instructions and Makefile enhancements

### PR DESCRIPTION
**Pull request by jhrozek**
This PR extends the readme with instructions on how to build content and push the content image to the cluster. It also adds a new Dockerfile.dev file which is a very simple Dockerfile to just consume a content XML built locally.

### CODE REVIEW

**Comment by jhrozek** [comment]
cc @JAORMX @mrogers950

File: Makefile [hunk 1]
```diff
@@ -1,29 +1,90 @@
...
+	@echo "Pushing image $(REPO)/$(APP_NAME):$(TAG) to the image registry"
+	IMAGE_REGISTRY_HOST=$$(oc get route default-route -n openshift-image-registry --template='{{ .spec.host }}'); \
+		$(CNT_RUNTIME) login --tls-verify=false -u $(OPENSHIFT_USER) -p $(shell oc whoami -t) $${IMAGE_REGISTRY_HOST}; \
+		$(CNT_RUNTIME) push --tls-verify=false $(FROM) $${IMAGE_REGISTRY_HOST}/$(NAMESPACE)/$(TO);
+	@echo "Removing the route from the image registry"
+	@oc patch configs.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":false}}' --type=merge
+	$(eval IMAGE_PATH = image-registry.openshift-image-registry.svc:5000/$(NAMESPACE)/$(APP_NAME):$(TAG))
+
+.PHONY: check-from
+check-from:
```
Comment (JAORMX) [inline Makefile:71]: where is this used?

File: Makefile [hunk 2]
```diff
@@ -1,29 +1,90 @@
...
+ifndef FROM
+	$(error FROM is undefined. See README.md for examples)
+endif
+
+.PHONY: check-to
+check-to:
```
Comment (JAORMX) [inline Makefile:77]: where is this used?

File: Makefile (hunk 1 above)
Comment (jhrozek) [reply Makefile:71]: Now it's used as a dependency of the image-to-cluster target. Previously I forgot to add it as a dep..

File: Makefile (hunk 2 above)
Comment (jhrozek) [reply Makefile:77]: Same answer as check-to

File: Makefile (hunk 1 above)
Comment (JAORMX) [reply Makefile:71]: cool, thanks for adding it.

File: Makefile [hunk 3]
```diff
@@ -1,29 +1,90 @@
...
 
+.PHONY: build-nocache
 build-nocache:
 	$(CNT_RUNTIME) build --build-arg=repo=${CONTENT_REPO} --build-arg=branch=${CONTENT_BRANCH} --no-cache -f Dockerfile -t $(APP_NAME)
 
+.PHONY: build-dev
+build-dev: check-content-path
+	$(eval CONTENT_NAME = $(notdir ${CONTENT_PATH}))
+	cp $(CONTENT_PATH) $(CONTENT_NAME)
+	-$(CNT_RUNTIME) build --build-arg=content=${CONTENT_NAME} -f Dockerfile.dev -t $(APP_NAME)
```
Comment (JAORMX) [inline Makefile]: why not add a tag already here?

Comment (jhrozek) [reply Makefile]: Good call. I added `:$(TAG)` to all the `build-*` targets and amended the Makefile a little.

**Review by JAORMX — APPROVED** [review]
worked for me!

**Pull request merged**